test/CMakeLists.txt

```diff
@@ -93,6 +93,14 @@ function(add_test)
   if(TEST_PROPERTIES)
     set_tests_properties(${test_name} PROPERTIES ${TEST_PROPERTIES})
   endif()
+
+  set(dll_command ${CMAKE_COMMAND} -E copy_if_different
+      $<TARGET_RUNTIME_DLLS:${target}> $<TARGET_FILE_DIR:${target}>)
+  add_custom_command(
+    TARGET ${target} POST_BUILD
+    COMMAND "$<$<BOOL:$<TARGET_RUNTIME_DLLS:${target}>>:${dll_command}>"
+    COMMAND_EXPAND_LISTS
+  )
```

Comment:
Since you are raising the CMake version for testing, just use the [ENVIRONMENT_MODIFICATION](https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT_MODIFICATION.html) property to append the DLLs path to PATH.

Reply:
I'll look into that, I wasn't aware of that feature.